src/envoy/mixer/grpc_transport.cc

```diff
@@ -72,6 +82,18 @@ bool GrpcTransport::IsMixerServerConfigured(Upstream::ClusterManager& cm) {
   return cm.get(kMixerServerClusterName) != nullptr;
 }
 
+void GrpcTransport::onPreRequestCustomizeHeaders(Http::HeaderMap& headers) {
+  if (!headers_) return;
+
+  for (const auto& key : kHttpTraceHeaders) {
+    const HeaderEntry* entry = headers_->get(key);
```

Comment:
It's in the context
https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-metadata.md

But at least we don't get the optimization for most of the inline headers. Is that OK? `get` is a linear search.